packages/firebase_database/lib/src/database_reference.dart

```diff
@@ -125,10 +128,65 @@ class DatabaseReference extends Query {
   ///
   /// remove() is equivalent to calling set(null)
   Future<Null> remove() => set(null);
+
+  /// Determine the next transaction key to be used. 0 is used if there are
+  /// no existing keys, otherwise the last key is incremented by 1.
+  int _getNextTransactionKey() {
+    if (FirebaseDatabase._transactions.isEmpty) {
+      return 0;
+    }
+    final int lastKey = FirebaseDatabase._transactions.keys.last;
+    return lastKey + 1;
+  }
+
+  /// Performs an optimistic-concurrency transactional update to the data at
+  /// this Firebase Database location.
+  Future<Null> runTransaction(TransactionHandler transactionHandler,
+      [int transactionTimeout = _transactionTimeout]) async {
+    if (transactionTimeout < 0) {
```

Comment:
I think it would be more concise to use an `assert(transactionTimeout > 0)` rather than throwing `ArgumentError` and in any case a timeout of zero doesn't make sense and should be disallowed.

Reply:
Removed default value in favor of checking for a null duration, in which case the default is applied.